test/auth/auto_detect_spec.js

```diff
@@ -0,0 +1,45 @@
+/* jshint mocha:true */
+var assert = require('assert');
+var sinon = require('sinon');
+var rewire = require('rewire');
+var autoDetect = rewire('../../lib/auth/auto_detect');
+var NativeFlow = require('../../lib/auth/native_flow');
+var RedirectFlow = require('../../lib/auth/redirect_flow');
+var ChromeExtensionFlow = require('../../lib/auth/chrome_extension_flow');
+var Environment = require('../../lib/environment');
+
+describe('autoDetect', function() {
+  var sandbox;
+
+  beforeEach(function(){
+    sandbox = sinon.sandbox.create();
```

Comment:
Sandbox seems like a nice way to isolate things between tests. Unclear if its necessary thought.